ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -113,41 +122,52 @@ public void execute(Input input, ActionListener<Output> listener) {
                             MLMemorySpec memorySpec = mlAgent.getMemory();
                             String memoryId = inputDataSet.getParameters().get(MEMORY_ID);
                             String parentInteractionId = inputDataSet.getParameters().get(PARENT_INTERACTION_ID);
+                            String regenerateInteractionId = inputDataSet.getParameters().get(REGENERATE_INTERACTION_ID);
                             String appType = mlAgent.getAppType();
                             String question = inputDataSet.getParameters().get(QUESTION);
 
+                            if (memoryId == null && regenerateInteractionId != null) {
+                                listener.onFailure(new MLValidationException("Memory id must provide for regenerate"));
+                            }
+
                             if (memorySpec != null
                                 && memorySpec.getType() != null
                                 && memoryFactoryMap.containsKey(memorySpec.getType())
                                 && (memoryId == null || parentInteractionId == null)) {
                                 ConversationIndexMemory.Factory conversationIndexMemoryFactory =
                                     (ConversationIndexMemory.Factory) memoryFactoryMap.get(memorySpec.getType());
-                                conversationIndexMemoryFactory.create(question, memoryId, appType, ActionListener.wrap(memory -> {
+                                conversationIndexMemoryFactory.create(question, memoryId, appType, wrap(memory -> {
```

Comment:
Keep the existing style of using ActionListener.wrap()? You only changed this wrap, but there are still some ActionListener.wrap() and ActionListener.* somewhere else in this file.

Reply:
this is unintentional change, change it back.